### PR TITLE
feat: v2 teams event types

### PR DESCRIPTION
**Pull request by supalarry**
Linear [CAL-4831](https://linear.app/calcom/issue/CAL-4831/platform-feat-teams-event-types)

### CODE REVIEW

File: apps/api/v2/src/modules/teams/memberships/controllers/teams-memberships.controller.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { Roles } from "@/modules/auth/decorators/roles/roles.decorator";
 import { ApiAuthGuard } from "@/modules/auth/guards/api-auth/api-auth.guard";
 import { RolesGuard } from "@/modules/auth/guards/roles/roles.guard";
 import { OrganizationsEventTypesService } from "@/modules/organizations/services/event-types/organizations-event-types.service";
```
Comment (SomayChauhan) [inline apps/api/v2/src/modules/teams/memberships/controllers/teams-memberships.controller.ts:5]: we can remove this, not used anywhere

**Review by ibex088 — CHANGES_REQUESTED** [review]
Minor nitpicks, otherwise looks great.

File: apps/api/v2/src/modules/teams/event-types/services/teams-event-types.service.ts [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+import { UsersService } from "@/modules/users/services/users.service";
+import { UserWithProfile } from "@/modules/users/users.repository";
+import { Injectable, NotFoundException, Logger } from "@nestjs/common";
+
+import { createEventType, updateEventType } from "@calcom/platform-libraries";
+import { InputTeamEventTransformed_2024_06_14 } from "@calcom/platform-types";
+
+@Injectable()
+export class TeamsEventTypesService {
+  private readonly logger = new Logger("OrganizationsEventTypesService");
```
[the remark this answers is not in the document]
Comment (supalarry) [reply apps/api/v2/src/modules/teams/event-types/services/teams-event-types.service.ts]: Fixed in [refactor: logger name and unused import](https://github.com/calcom/cal.com/pull/18085/commits/6bf1ac90874af61b54b51469c34549b477804634)

File: apps/api/v2/src/modules/teams/memberships/controllers/teams-memberships.controller.ts (hunk 1 above)
Comment (supalarry) [reply apps/api/v2/src/modules/teams/memberships/controllers/teams-memberships.controller.ts]: Fixed in [refactor: logger name and unused import](https://github.com/calcom/cal.com/pull/18085/commits/6bf1ac90874af61b54b51469c34549b477804634)

**Review by ibex088 — APPROVED** [review]

**Pull request merged**